rasa/server.py

```diff
@@ -131,7 +139,7 @@ def decorated(*args, **kwargs):
 def requires_auth(app: Sanic, token: Optional[Text] = None) -> Callable[[Any], Any]:
     """Wraps a request handler with token authentication."""
 
-    def decorator(f: Callable[[Any, Any], Any]) -> Callable[[Any, Any], Any]:
+    def decorator(f: "SanicView") -> "SanicView":
```

Comment:
why do we need to define a `SanicView` here? (that concludes my review)

Reply:
since we're manipulating arguments inside the decorator and call `f` with `request`, we need the type annotation to tell mypy that `f` is a function that accepts `request` as first argument. Does that make sense?

Reply:
Yes, it does. And something like `Callable[[Request, ??], BaseHTTPResponse]` doesn't work as we only know the first argument, but not the others, right?

Reply:
🤔 will try `Callable[[Request, ...], BaseHTTPResponse]`

Reply:
I was able to make it work with 297c1dcca4a (it's admittedly simpler)